server/api/api/endpoints/nuclio.py

```diff
@@ -136,7 +136,17 @@ async def store_api_gateway(
                 for func in existing_api_gateway.get_function_names()
                 if func not in api_gateway.get_function_names()
             ]
-            if unused_functions:
+            # if invocation URL has changed, delete URL from all the functions
+            # if only function list has changed, then delete URL only from those functions
+            # which are not used in api gateway anymore
```

Comment:
it's irrelevant for this condition, move this comment above `if unused_function`